RecoLocalCalo/HcalRecProducers/plugins/alpaka/HBHERecHitProducerPortable.cc

```diff
@@ -0,0 +1,169 @@
+#include "CondFormats/HcalObjects/interface/alpaka/HcalMahiConditionsDevice.h"
+#include "CondFormats/DataRecord/interface/HcalRecoParamsRcd.h"
+#include "CondFormats/HcalObjects/interface/alpaka/HcalRecoParamWithPulseShapeDevice.h"
+#include "CondFormats/HcalObjects/interface/alpaka/HcalMahiPulseOffsetsDevice.h"
+
+#include "DataFormats/HcalDigi/interface/alpaka/HcalDigiDeviceCollection.h"
+#include "DataFormats/HcalRecHit/interface/alpaka/HcalRecHitDeviceCollection.h"
+
+#include "HeterogeneousCore/AlpakaInterface/interface/config.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/Event.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EventSetup.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDGetToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/EDPutToken.h"
+#include "HeterogeneousCore/AlpakaCore/interface/alpaka/stream/EDProducer.h"
+#include "HeterogeneousCore/CUDACore/interface/JobConfigurationGPURecord.h"
+
+#include "Mahi.h"
+#include "CalibCalorimetry/HcalAlgos/interface/HcalTimeSlew.h"
+
+namespace ALPAKA_ACCELERATOR_NAMESPACE {
+
+  class HBHERecHitProducerPortable : public stream::EDProducer<> {
+  public:
+    explicit HBHERecHitProducerPortable(edm::ParameterSet const&);
+    ~HBHERecHitProducerPortable() override;
+    static void fillDescriptions(edm::ConfigurationDescriptions&);
+
+  private:
+    void produce(device::Event&, device::EventSetup const&) override;
+
+    using IProductTypef01 = hcal::Phase1DigiDeviceCollection;
+    const device::EDGetToken<IProductTypef01> digisTokenF01HE_;
+
+    using IProductTypef5 = hcal::Phase0DigiDeviceCollection;
+    const device::EDGetToken<IProductTypef5> digisTokenF5HB_;
+
+    using IProductTypef3 = hcal::Phase1DigiDeviceCollection;
+    const device::EDGetToken<IProductTypef3> digisTokenF3HB_;
+
+    using OProductType = hcal::RecHitDeviceCollection;
+    const device::EDPutToken<OProductType> rechitsM0Token_;
+
+    const device::ESGetToken<hcal::HcalMahiConditionsPortableDevice, HcalMahiConditionsRcd> mahiConditionsToken_;
+    const device::ESGetToken<HcalSiPMCharacteristicsPortableDevice, HcalSiPMCharacteristicsRcd>
+        sipmCharacteristicsToken_;
+    const device::ESGetToken<HcalRecoParamWithPulseShapeDevice, HcalRecoParamsRcd> recoParamsToken_;
+    const device::ESGetToken<HcalMahiPulseOffsetsPortableDevice, JobConfigurationGPURecord> mahiPulseOffsetsToken_;
+    //
+
+    hcal::reconstruction::ConfigParameters configParameters_;
+  };
+
+  HBHERecHitProducerPortable::HBHERecHitProducerPortable(edm::ParameterSet const& ps)
+      : digisTokenF01HE_{consumes(ps.getParameter<edm::InputTag>("digisLabelF01HE"))},
+        digisTokenF5HB_{consumes(ps.getParameter<edm::InputTag>("digisLabelF5HB"))},
+        digisTokenF3HB_{consumes(ps.getParameter<edm::InputTag>("digisLabelF3HB"))},
+        rechitsM0Token_{produces(ps.getParameter<std::string>("recHitsLabelM0HBHE"))},
```

Comment:
Please remove the `recHitsLabelM0HBHE` parameter and use an empty label.

Reply:
Done in 6ba6f2400bad40a6f2c057fda1e488559181d93a